src/Symfony/Component/Cache/Traits/RedisTrait.php

```diff
@@ -174,28 +174,24 @@ public static function createConnection(#[\SensitiveParameter] string $dsn, arra
 
         $params += $query + $options + self::$defaultConnectionOptions;
 
-        if (isset($params['redis_sentinel']) && isset($params['sentinel_master'])) {
-            throw new InvalidArgumentException('Cannot use both "redis_sentinel" and "sentinel_master" at the same time.');
-        }
-
-        $params['redis_sentinel'] ??= $params['sentinel_master'] ?? null;
+        $params['sentinel'] ??= $params['sentinel_master'] ?? $params['redis_sentinel'] ?? null;
 
-        if (isset($params['redis_sentinel']) && !class_exists(\Predis\Client::class) && !class_exists(\RedisSentinel::class) && !class_exists(Sentinel::class)) {
+        if (isset($params['sentinel']) && !class_exists(\Predis\Client::class) && !class_exists(\RedisSentinel::class) && !class_exists(Sentinel::class)) {
             throw new CacheException('Redis Sentinel support requires one of: "predis/predis", "ext-redis >= 5.2", "ext-relay".');
         }
 
         if (isset($params['lazy'])) {
             $params['lazy'] = filter_var($params['lazy'], \FILTER_VALIDATE_BOOLEAN);
         }
-        $params['redis_cluster'] = filter_var($params['redis_cluster'], \FILTER_VALIDATE_BOOLEAN);
+        $params['cluster'] = filter_var($params['cluster'] ?? $params['redis_cluster'] ?? null, \FILTER_VALIDATE_BOOLEAN);
```

Comment:
as `$params` has merged the default options, which have a `false` default for `cluster`, this will never work for `redis_cluster`